myhoard/controller.py

```diff
@@ -1174,7 +1174,9 @@ def _handle_mode_restore(self):
         self._purge_old_binlogs(mysql_maybe_not_running=True)
         self._process_local_binlog_updates()
         self._extend_binlog_stream_list()
-        if self.restore_coordinator.phase == RestoreCoordinator.Phase.failed_basebackup:
+        if self.restore_coordinator.phase is RestoreCoordinator.Phase.failed_disk_full:
+            self._switch_basebackup_if_possible()
```

Comment:
Trying another backup isn't probably very useful. Since we don't want the controller to do anything special in this case we could omit the new restore coordinator state and set the state in `RestoreCoordinator` to `RestoreCoordinator.Phase.failed` when encountering `DiskFullError`

Reply:
The question was more... "maybe the last one is too big, but an older one is small enough to be restored to?"

But I'm okay with just failing and not trying other basebackups.

WDYT?

Reply:
If we restore an older backup all the binlogs from both the old and new backup are still applied and it ends up with the same data. Maybe there is some theoretical case where old basebackup is smaller and the binlogs won't make it as big as the more recent basebackup itself but I don't really see how that would happen.